framework/src/systems/SystemBase.C

```diff
@@ -1433,7 +1451,11 @@ SystemBase::needSolutionState(const unsigned int state)
   // We will manually add all states past current
   for (unsigned int i = 1; i <= state; ++i)
     if (!_solution_states[i])
-      _solution_states[i] = &addVector(oldSolutionStateVectorName(i), true, GHOSTED);
+    {
+      auto tag =
+          _subproblem.addVectorTag(oldSolutionStateVectorName(i), Moose::VECTOR_TAG_SOLUTION);
```

Comment:
I tagged old and older solutions because we are managing them but we cannot tag the solution `solution()` or `currentSolution()` because it is a vector passed in by the solver and could be changing. This causes the special treatment in `coupledVectorTagValue` when the tag name is `SOLUTION_TAG`.

Reply:
We should stay with `currentSolution()` that is managed in libmesh not in solver side.  It does not change  within one one time step

Reply:
It is true on libmesh side, but moose manage it as a `const` solution pointer. So we cannot directly use that pointer for tagging because tagging assumes we want to change a vector. I believe that this pointer is not always pointing to libmesh current solution. I tried to directly tag that vector but failed. We possibly should tag during calling `setSolution` but in that function the solution is passed in as a const reference which we cannot tag unless we change our current design in `SystemBase.h`:
```
std::vector<NumericVector<Number> *> _tagged_vectors;
```